From 0baf0188b5ddd7afc1ec6512729029214f1a2ac4 Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@st.com>
Date: Mon, 22 Oct 2018 13:54:24 +0200
Subject: [PATCH 3/3] 4.19: correct .ndo_select_queue pointer function cast

Since kernel 4.19, the parameters list of function 'ndo_select_queue'
have changed.

Signed-off-by: Christophe Priouzeau <christophe.priouzeau@st.com>
---
 os_dep/os_intfs.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/os_dep/os_intfs.c b/os_dep/os_intfs.c
index 44cd53b..e3ab4f2 100644
--- a/os_dep/os_intfs.c
+++ b/os_dep/os_intfs.c
@@ -581,7 +581,9 @@ static unsigned int rtw_classify8021d(struct sk_buff *skb)
 }
 
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 0)
+			, struct net_device *sb_dev
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0)
 			, void *accel_priv
 #endif // kernle >= 3.13
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
-- 
2.7.4

